ledger/accountdb_test.go

```diff
@@ -1069,6 +1080,303 @@ func BenchmarkWriteCatchpointStagingBalances(b *testing.B) {
+				[]byte("DingHo-SmallPack"),
+				[]byte("DingHo-StandardPack"),
+			},
+		},
+		{
+			prefix: []byte("DingHo-S"),
+			expectedNames: [][]byte{
+				[]byte("DingHo-SmallPack"),
+				[]byte("DingHo-StandardPack"),
+			},
+		},
+		{
+			prefix:        []byte("DingHo-Small"),
+			expectedNames: [][]byte{[]byte("DingHo-SmallPack")},
+		},
+		{
+			prefix:        []byte("BostonKitchen"),
+			expectedNames: [][]byte{[]byte("BostonKitchen-CheeseSlice")},
+		},
+		{
+			prefix:        []byte(`™£´´∂ƒ∂ƒßƒ©`),
+			expectedNames: [][]byte{[]byte(`™£´´∂ƒ∂ƒßƒ©∑®ƒß∂†¬∆`)},
+		},
+		{
+			prefix: []byte{},
+			expectedNames: [][]byte{
+				{0xFF, 0x12, 0x34, 0x56, 0x78},
+				{0xFF, 0xFF, 0x34, 0x56, 0x78},
+				{0xFF, 0xFF, 0xFF, 0x56, 0x78},
+				{0xFF, 0xFF, 0xFF, 0xFF, 0x78},
+				{0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+				{0xFF, 0xFE, 0xFF},
+				{0xFF, 0xFF},
+				{0xBA, 0xDD, 0xAD, 0xFF, 0xFF},
+				{0xBA, 0xDD, 0xAE, 0x00},
+				{0xBA, 0xDD, 0xAE},
+				[]byte("TACOCAT"),
+				[]byte("TACOBELL"),
+				[]byte("DingHo-SmallPack"),
+				[]byte("DingHo-StandardPack"),
+				[]byte("BostonKitchen-CheeseSlice"),
+				[]byte(`™£´´∂ƒ∂ƒßƒ©∑®ƒß∂†¬∆`),
+			},
+		},
+	}
+
+	for index, testCase := range testCases {
+		t.Run("lookupKVByPrefix-testcase-"+strconv.Itoa(index), func(t *testing.T) {
+			actual := make(map[string]bool)
+			_, err := qs.lookupKeysByPrefix(string(testCase.prefix), uint64(len(kvPairDBPrepareSet)), actual, 0)
+			require.NoError(t, err)
+			expected := make(map[string]bool)
+			for _, name := range testCase.expectedNames {
+				expected[string(name)] = true
+			}
+			require.Equal(t, actual, expected)
+		})
+	}
+}
+
+func BenchmarkLookupKeyByPrefix(b *testing.B) {
+	// learn something from BenchmarkWritingRandomBalancesDisk
+
+	dbs, fn := dbOpenTest(b, false)
+	setDbLogging(b, dbs)
+	defer cleanupTestDb(dbs, fn, false)
+
+	// return account data, initialize DB tables from accountsInitTest
+	_ = benchmarkInitBalances(b, 1, dbs, protocol.ConsensusCurrentVersion)
+
+	qs, err := accountsInitDbQueries(dbs.Rdb.Handle)
+	require.NoError(b, err)
+	defer qs.close()
+
+	currentDBSize := 0
+	nextDBSize := 2
+	increment := 2
+
+	nameBuffer := make([]byte, 5)
+	valueBuffer := make([]byte, 5)
+
+	// from 2^1 -> 2^2 -> ... -> 2^22 sized DB
+	for bIndex := 0; bIndex < 22; bIndex++ {
+		// make writer to DB
+		tx, err := dbs.Wdb.Handle.Begin()
+		require.NoError(b, err)
+
+		// writer is only for kvstore
+		writer, err := makeAccountsSQLWriter(tx, true, true, true, true)
+		if err != nil {
+			return
+		}
+
+		var prefix string
+		// how to write to dbs a bunch of stuffs?
+		for i := 0; i < nextDBSize-currentDBSize; i++ {
+			crypto.RandBytes(nameBuffer)
+			crypto.RandBytes(valueBuffer)
+			appID := basics.AppIndex(crypto.RandUint64())
+			boxKey := logic.MakeBoxKey(appID, string(nameBuffer))
+			err = writer.upsertKvPair(boxKey, string(valueBuffer))
+			require.NoError(b, err)
+
+			if i == 0 {
+				prefix = logic.MakeBoxKey(appID, "")
+			}
+		}
+		tx.Commit()
```

Comment:
minor comment in testing:
```suggestion
		err = tx.Commit()
		require.NoError(b, err)
```